app/controllers/CsvFileUploadController.scala

```diff
@@ -176,6 +179,29 @@ class CsvFileUploadController @Inject()(val mcc: MessagesControllerComponents,
     }
   }
 
+  def checkFileNames(csvCallbackData: List[UploadedSuccessfully], schemeInfo: SchemeInfo)(implicit authContext: ERSAuthData,
+                                                                                          request: Request[AnyRef],
+                                                                                          hc: HeaderCarrier): Future[Result] = {
+    ersUtil.fetch[UpscanCsvFilesList](ersUtil.CSV_FILES_UPLOAD, schemeInfo.schemeRef).flatMap { list =>
+      val uploadedWithCorrectName: Boolean = list.ids
+        .map(expectedFile => expectedFile.fileId)
+        .zip(csvCallbackData.reverse
+          .map(uploadedFile => uploadedFile.name))
+        .map(t => (ersUtil.getPageElement(schemeInfo.schemeId, ersUtil.PAGE_CHECK_CSV_FILE, t._1 + ".file_name"), t._2))
```

Comment:
could you split this tuple up?
could do
```.map{ (expectedName, uploadedName) => ...}```
 or, if that is not allowed for some reason,
```
.map{ names => 
    val expectedName = names._1
    val uploadedName = names._2
    ...
}
```